EpikCord/__init__.py

```diff
@@ -1156,6 +1113,151 @@ async def command_error(
+        return self.guilds.fetch(data["guild_id"])
+
+    async def guild_member_add(self,data:dict) -> tuple[Optional[Guild], GuildMember]:
+        return self.guilds.fetch(data["guild_id"]), GuildMember(self, data)    
+
+    async def guild_member_update(self, data) -> tuple[Optional[GuildMember],GuildMember]:
+        guild_member = GuildMember(self, data)
+        return self.members.fetch(data["id"]), guild_member
+
+    async def guild_member_remove(self, data:dict) -> tuple[Optional[Guild], User]:
+        return self.guilds.fetch(data["guild_id"]), User(self,data["user"])
+
+    async def guild_members_chunk(self, data: dict):
+        ...
+    
+    # Guild Role: TODO: Cache Roles per Server
+    async def guild_role_create(self,data:dict) -> Role:
+        role = Role(self,data["role"])
+        self.roles.add_to_cache(role.id, role)
+        return role
+
+    async def guild_role_update(self,data:dict) -> tuple[Optional[Role], Role]:
+        after = Role(self, data["role"])
+        before = self.roles.get(after.id)
+        return before, after 
+
+    async def guild_role_delete(self, data:dict) -> Optional[Role]:
+        deleted = self.roles.get(data["role_id"])
+        return deleted
+    
+    # Guild Scheduled Event Create: TODO: Caching
+    async def guild_scheduled_event_create(self, data:dict) -> GuildScheduledEvent:
+        event = GuildScheduledEvent(self, data)
+        self.guild_events.add_to_cache(event.id, event)
+        return event
+    
+    async def guild_scheduled_event_update(self,data:dict):
+        after = GuildScheduledEvent(self,data)
+        before = self.guild_events.get(after.id)
+        return before, after
+    async def guild_scheduled_event_delete(self,data:dict):
+        return GuildScheduledEvent(self, data)
+
+    async def guild_scheduled_event_user_add(self, data:dict):
+        ...
+    # Channel Events
+    async def channel_create(self, data: dict)-> AnyChannel:
+        channel = self.utils.channel_from_type(data)
+        self.channels.add_to_cache(channel.id, channel)
+        return channel
+    
+    async def channel_update(self, data:dict) -> tuple[Optional[AnyChannel], AnyChannel]:
+        after:GuildChannel = self.utils.channel_from_type(data)
+        before = self.channels.get(after.id)
+        return before,after
+    
+    async def channel_delete(self,data:dict)-> Optional[AnyChannel]:
+        return self.channels.get(data["id"])
+    
+    async def channel_pins_update(self,data:dict) -> tuple[Optional[str], str, Optional[datetime.datetime]]:
+        timestamp = data.get("last_pin_timestamp")
+        dt = datetime.datetime.fromisoformat(timestamp) if timestamp else None
+        return (data.get("guild_id"), data["channel_id"], dt)
+    
+    # Thread Events, TODO: Need to do more Events 
+    async def thread_list_sync(self,data:dict):
+        return ThreadSyncEvent(self, data)
+
+    # Message Events
+    async def message_create(self, data: dict)-> Message:
```

Comment:
pls just stop using `dict` annotation...

Reply:
Why not